AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSCognitoUserPoolTokens.swift

```diff
@@ -38,6 +39,38 @@ public struct AWSCognitoUserPoolTokens: AuthCognitoTokens {
         self.expiration = expiration
     }
 
+    public init(idToken: String,
+                accessToken: String,
+                refreshToken: String,
+                expiresIn: Int? = nil) {
+
+        if let expiresIn =  expiresIn {
+            self.init(idToken: idToken,
+                      accessToken: accessToken,
+                      refreshToken: refreshToken,
+                      expiresIn: expiresIn)
+        } else {
+
+            var expirationDoubleValue: Double? = nil
+            if let idTokenExpiration = try? AWSAuthService().getTokenClaims(tokenString: idToken).get()["exp"]?.doubleValue {
+                expirationDoubleValue = idTokenExpiration
+            }
+            if let accessTokenExpiration = try? AWSAuthService().getTokenClaims(tokenString: accessToken).get()["exp"]?.doubleValue {
+                if let unwrappedExpirationDoubleValue = expirationDoubleValue {
+                    expirationDoubleValue = min(unwrappedExpirationDoubleValue, accessTokenExpiration)
+                } else {
+                    expirationDoubleValue = accessTokenExpiration
+                }
+            }
```

Comment:
I think a switch statement might be easier to read, for example, i gave this a try and looks like this:
```swift
struct TwoOptionals {
    
    func returnDoubleOrNil() throws -> Double? {
        return nil
    }
    
    func example() -> Int {
        let finalValue: Double
        switch (try? returnDoubleOrNil(), try? returnDoubleOrNil()) {
        case (.some(let value1), .none):
            finalValue = value1
        case (.none, .some(let value2)):
            finalValue = value2
        case (.some(let value1), .some(let value2)):
            finalValue = min(value1, value2)
        case (.none, .none):
            finalValue = 0
        }
        return Int(finalValue.rounded(.down))
        
    }
}

```